### PR TITLE
Implement @empowerment.

### LINKED ISSUE #77: Decorator for installables
Something like this:
```
@empowers(IFoo)
class MyPowerup(Item):
    …
```
which would be equivalent to:
```
@implementer(IFoo)
class MyPowerup(Item):
    powerupInterfaces = [IFoo]
```

The main motivation for this is that it makes no sense to have an interface in `powerupInterfaces` that you do not implement. Note that you can still implement interfaces that you don't power up for:
```
@implementer(INotPowerupInterface)
@empowers(IFoo)
class MyPowerup(Item):
    …
```
works fine.

### PR DESCRIPTION
**Pull request by mithrandi**
Closes #77.

### CODE REVIEW

File: axiom/item.py [hunk 1]
```diff
@@ -1135,3 +1137,17 @@ class _PowerupConnector(Item):
 
 POWERUP_BEFORE = 1              # Priority for 'high' priority powerups.
 POWERUP_AFTER = -1              # Priority for 'low' priority powerups.
+
+
+
+def empowerment(iface, priority=0):
+    """
+    Class decorator for indicating a powerup's powerup interfaces.
```
Comment (jonathanj) [inline axiom/item.py]: I think this documentation should be a little more explanatory, particularly that it also declares the thing it’s decorating as implementing the interface.

Comment (jonathanj) [reply axiom/item.py]: And document the parameters.

File: axiom/test/test_item.py [hunk 2]
```diff
@@ -584,3 +585,68 @@ def txn():
...
+        @empowerment(TestInterface2, 20)
+        @empowerment(TestInterface)
+        class TI2(Item):
+            pass
+
+        self.assertEqual(
+            TI2()._getPowerupInterfaces(),
+            [(TestInterface, 0),
+             (TestInterface2, 20)])
+        self.assertTrue(TestInterface.implementedBy(TI2))
```
[the remark this answers is not in the document]
Comment (mithrandi) [reply axiom/test/test_item.py]: Oops, fixed!

File: axiom/test/test_item.py [hunk 3]
```diff
@@ -584,3 +585,68 @@ def txn():
...
+        """
+        @empowerment(TestInterface)
+        class TI3(Item):
+            powerupInterfaces = [(TestInterface2, 20)]
+
+        self.assertEqual(
+            TI3()._getPowerupInterfaces(),
+            [(TestInterface2, 20),
+             (TestInterface, 0)])
+        self.assertTrue(TestInterface.implementedBy(TI3))
```
[the remark this answers is not in the document]
Comment (mithrandi) [reply axiom/test/test_item.py]: I made the test explicitly implement it now, as would normally be done.

File: axiom/item.py (hunk 1 above)
Comment (mithrandi) [reply axiom/item.py]: Done.

**Pull request merged**